shm/src/Player.cpp

```diff
@@ -43,3 +43,13 @@ void Player::payCrew(const size_t payCrew) {
 void Player::setPlayerPtr() {
     ship_->changeDelegate(this);
 }
+
+void Player::buy(const size_t & amount, const size_t & price, const size_t & cargoNumber) {
+    money_ -= (amount * price);
+    ship_->load(getCargo(cargoNumber));
+}
```

Comment:
why not simple pass Cargo pointer:
```cpp
void Player::buy(std::shared_ptr<Cargo> cargo, size_t amount, size_t price) {
      money_ -= amount * price;
      ship_->load(cargo);
 }
```
there's no need to pass simple datatypes as const references (size_t is unsigned long underneath)
plus redundant brackets (totalPrice can be passed instead two arguments: amount * price can be calculated in Store)

Reply:
.